### PR TITLE
Captcha check added to Sign Up

### PR DESCRIPTION
**Pull request by swpinchen**
Implemented Google Captcha on Sign Up page.

Screenshot 2021-05-07 at 9 04 47

Updated gem file, so you'll need to do `bundle install` next time you do a `git pull`

### CODE REVIEW

File: app/controllers/users/confirmations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
```
Comment (Naokimi) [inline app/controllers/users/confirmations_controller.rb:1]: why this file? you should delete anything you aren't actively using

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
```
Comment (Naokimi) [inline app/controllers/users/omniauth_callbacks_controller.rb:1]: why this file? you should delete anything you aren't actively using

File: app/controllers/users/passwords_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
```
Comment (Naokimi) [inline app/controllers/users/passwords_controller.rb:1]: why this file? you should delete anything you aren't actively using

File: app/controllers/users/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
```
Comment (Naokimi) [inline app/controllers/users/registrations_controller.rb:1]: delete all the comments, keep only the `create` method

File: app/controllers/users/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
```
Comment (Naokimi) [inline app/controllers/users/sessions_controller.rb:1]: why this file? you should delete anything you aren't actively using

File: app/controllers/users/unlocks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
```
Comment (Naokimi) [inline app/controllers/users/unlocks_controller.rb:1]: why this file? you should delete anything you aren't actively using

File: config/environments/development.rb [hunk 7]
```diff
@@ -63,14 +63,8 @@
   config.action_mailer.default_url_options = { host: "http://localhost:3000" }
   config.action_mailer.perform_deliveries = true
   config.action_mailer.delivery_method = :test
-  # config.action_mailer.delivery_method = :smtp
```
Comment (Naokimi) [inline config/environments/development.rb:66]: before deleting this, have you confirmed that emails are sent from development without it?

File: config/routes.rb [hunk 8]
```diff
@@ -1,12 +1,20 @@
 Rails.application.routes.draw do
   get 'users/show'
   get 'settings/change', to: "settings#change"
-  devise_for :users
+
+  # Controller for captcha
+  devise_for :users, controllers: {
+    :registrations => 'users/registrations'
```
Comment (Naokimi) [inline config/routes.rb:7]: this is the old way of writing, please try to keep as much as possible to the new way `registrations: 'users/registrations'`
also, if it's a singe key-value pair I'd personally keep it inline

**Review by Naokimi — CHANGES_REQUESTED** [review]
no 
```
  RECAPTCHA_SITE_KEY= ENV['RECAPTCHA_SITE_KEY']
  RECAPTCHA_SECRET_KEY= ENV['RECAPTCHA_SECRET_KEY']
```
for staging environment?

File: app/controllers/users/confirmations_controller.rb (hunk 1 above)
Comment (swpinchen) [reply app/controllers/users/confirmations_controller.rb]: Unnecessary file created by `rails g devise:controllers users`.  Now deleted.  Thanks!

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (swpinchen) [reply app/controllers/users/omniauth_callbacks_controller.rb]: Unnecessary file created by `rails g devise:controllers users`.  Now deleted.  Thanks!

File: app/controllers/users/passwords_controller.rb (hunk 3 above)
Comment (swpinchen) [reply app/controllers/users/passwords_controller.rb]: Unnecessary file created by `rails g devise:controllers users`.  Now deleted.  Thanks!

File: app/controllers/users/registrations_controller.rb (hunk 4 above)
Comment (swpinchen) [reply app/controllers/users/registrations_controller.rb:1]: Comments deleted.  Thanks!

File: app/controllers/users/sessions_controller.rb (hunk 5 above)
Comment (swpinchen) [reply app/controllers/users/sessions_controller.rb]: Unnecessary file created by `rails g devise:controllers users`.  Now deleted.  Thanks!

File: app/controllers/users/unlocks_controller.rb (hunk 6 above)
Comment (swpinchen) [reply app/controllers/users/unlocks_controller.rb]: Unnecessary file created by `rails g devise:controllers users`.  Now deleted.  Thanks!

**Review by Naokimi — APPROVED** [review]

File: config/environments/development.rb (hunk 7 above)
Comment (swpinchen) [reply config/environments/development.rb:66]: Thanks Paulo.  These lines were already commented when mail delivery was tested in dev by confirming the mail message output to rails server.  At some time in the future, perhaps we can add the same configuration as production and catch outgoing emails with mail catcher.

File: config/routes.rb (hunk 8 above)
Comment (swpinchen) [reply config/routes.rb]: @Naokimi Thanks for your help Paulo.  I agree with all your suggestions and learned a lot!

File: app/controllers/users/registrations_controller.rb (hunk 4 above)
Comment (swpinchen) [reply app/controllers/users/registrations_controller.rb:1]: @Naokimi At the moment our Staging environment uses the same configuration as production.

**Pull request merged**